### PR TITLE
DataGrid: Extract row_based module from editing

### PR DESCRIPTION
**Pull request by jtoming830**

### CODE REVIEW

File: js/ui/grid_core/ui.grid_core.editing.js [hunk 1]
```diff
@@ -1428,7 +1380,7 @@ const EditingController = modules.ViewController.inherit((function() {
         _processSaveEditDataResult: function(results) {
             const that = this;
             let hasSavedData = false;
-            const editMode = getEditMode(that);
+            const editMode = that.getEditMode();
```
Comment (NickMitrokhin) [inline js/ui/grid_core/ui.grid_core.editing.js:1383]: We should get rid of "that". Use "this" instead.

File: js/ui/grid_core/ui.grid_core.editing.js [hunk 2]
```diff
@@ -903,7 +874,7 @@ const EditingController = modules.ViewController.inherit((function() {
             const oldEditRowIndex = that._getVisibleEditRowIndex();
             const change = { data, type: DATA_EDIT_DATA_INSERT_TYPE, parentKey };
             const { key, rowIndex } = that._addInsertInfo(change);
-            const editMode = getEditMode(that);
+            const editMode = that.getEditMode();
```
Comment (NickMitrokhin) [inline js/ui/grid_core/ui.grid_core.editing.js:877]: We should get rid of "that". Use "this" instead.

File: js/ui/grid_core/ui.grid_core.editing.js [hunk 3]
```diff
@@ -1649,16 +1595,7 @@ const EditingController = modules.ViewController.inherit((function() {
...
-                const editMode = getEditMode(that);
-                let isVisibleWithCurrentEditMode = false;
-
-                if(editMode === EDIT_MODE_ROW) {
-                    isVisibleWithCurrentEditMode = editingOptions.allowUpdating || editingOptions.allowAdding;
-                }
-
-                return editingOptions.allowDeleting || isVisibleWithCurrentEditMode;
-            }
+            return editingOptions.allowDeleting;
```
Comment (NickMitrokhin) [inline js/ui/grid_core/ui.grid_core.editing.js:1598]: We should get rid of "that" in this method. Use "this" instead.

File: js/ui/grid_core/ui.grid_core.editing.js [hunk 4]
```diff
@@ -1846,7 +1780,7 @@ const EditingController = modules.ViewController.inherit((function() {
         },
         _applyChange: function(options, params, forceUpdateRow) {
             const that = this;
-            const editMode = getEditMode(that);
+            const editMode = that.getEditMode();
```
Comment (NickMitrokhin) [inline js/ui/grid_core/ui.grid_core.editing.js:1783]: We should get rid of "that". Use "this" instead.

File: js/ui/grid_core/ui.grid_core.editing.js [hunk 5]
```diff
@@ -1973,15 +1914,15 @@ const EditingController = modules.ViewController.inherit((function() {
             const column = options.column;
             const rowIndex = options.row && options.row.rowIndex;
             let template;
-            const isRowMode = isRowEditMode(that);
+            const isRowMode = this.isRowBasedEditMode();
```
Comment (NickMitrokhin) [inline js/ui/grid_core/ui.grid_core.editing.js:1917]: We should get rid of "that" in this method. Use "this" instead.

File: js/ui/grid_core/ui.grid_core.editing.js [hunk 6]
```diff
@@ -2109,7 +2050,7 @@ const EditingController = modules.ViewController.inherit((function() {
                 buttonItems.push(prepareButtonItem('addRow', 'addRow', 20));
             }
 
-            if((editingOptions.allowUpdating || editingOptions.allowAdding || editingOptions.allowDeleting) && getEditMode(that) === EDIT_MODE_BATCH) {
+            if((editingOptions.allowUpdating || editingOptions.allowAdding || editingOptions.allowDeleting) && that.getEditMode() === EDIT_MODE_BATCH) {
```
Comment (NickMitrokhin) [inline js/ui/grid_core/ui.grid_core.editing.js:2053]: We should get rid of "that". Use "this" instead.

**Review by NickMitrokhin — APPROVED** [review]

**Review by vconst — APPROVED** [review]

**Pull request merged**